src/Microsoft.TemplateEngine.Cli/Commands/TemplateCommand.cs

```diff
@@ -259,5 +275,177 @@ private Option CreateChoiceOption(CliTemplateParameter parameter, IEnumerable<st
             return option;
         }
 
+        private ParseArgument<T> GetParseArgument<T>(CliTemplateParameter parameter, Func<string?, (bool, T)> convert)
+        {
+            return (argumentResult) =>
+            {
+                if (argumentResult.Parent is not OptionResult or)
+                {
+                    throw new NotSupportedException("The method should be only used with option.");
+                }
+
+                if (argumentResult.Tokens.Count == 0)
+                {
+                    if (or.IsImplicit)
+                    {
+                        if (parameter.DefaultValue != null)
+                        {
+                            (bool parsed, T value) = convert(parameter.DefaultValue);
+                            if (parsed)
+                            {
+                                return value;
+                            }
+                            argumentResult.ErrorMessage = $"Cannot parse default value '{parameter.DefaultValue}' for option '{or.Token?.Value}' as expected type {typeof(T).Name}.";
+                            //https://github.com/dotnet/command-line-api/blob/5eca6545a0196124cc1a66d8bd43db8945f1f1b7/src/System.CommandLine/Argument%7BT%7D.cs#L99-L113
+                            //TODO: system-command-line can handle null.
+                            return default!;
```

Comment:
@jonsequitur is it possible to change `ParseArgument` delegate to accept `null`?
I see in https://github.com/dotnet/command-line-api/blob/5eca6545a0196124cc1a66d8bd43db8945f1f1b7/src/System.CommandLine/Argument%7BT%7D.cs#L99-L113 it is possible to have null return here. Then `!` won't be needed.